### PR TITLE
Removes a spawn from traitor code

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code/game/gamemodes/traitor/traitor.dm [hunk 1]
```diff
@@ -60,13 +60,12 @@
 
 /datum/game_mode/traitor/post_setup()
 	for(var/datum/mind/traitor in pre_traitors)
-		spawn(rand(10,100))
-			traitor.add_antag_datum(antag_datum)
+		traitor.add_antag_datum(antag_datum)
```
Comment (Cyberboss) [inline code/game/gamemodes/traitor/traitor.dm]: use a timer

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/traitor/traitor.dm (hunk 1 above)
Comment (MoreRobustThanYou) [reply code/game/gamemodes/traitor/traitor.dm]: Why is a timer necessary for this?

File: code/game/gamemodes/traitor/traitor.dm [hunk 2]
```diff
@@ -60,13 +60,15 @@
 
 /datum/game_mode/traitor/post_setup()
 	for(var/datum/mind/traitor in pre_traitors)
-		spawn(rand(10,100))
-			traitor.add_antag_datum(antag_datum)
+		addtimer(CALLBACK(src, .proc/maketraitor, traitor), rand(30,60))
```
Comment (ChangelingRain) [inline code/game/gamemodes/traitor/traitor.dm]: why is this number different.

File: code/game/gamemodes/traitor/traitor.dm (hunk 1 above)
Comment (MoreRobustThanYou) [reply code/game/gamemodes/traitor/traitor.dm]: Ok, done.

File: code/game/gamemodes/traitor/traitor.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/gamemodes/traitor/traitor.dm]: Use the original numbers

Comment (Cyberboss) [reply code/game/gamemodes/traitor/traitor.dm]: Use a proper callback for add_antag_datum

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by AnturK** [comment]
Why is the delay there anyway ?

**Comment by Cyberboss** [comment]
For late joiners

On Oct 9, 2017 8:39 AM, "AnturK" <notifications@github.com> wrote:

Why is the delay there anyway ?

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/31465#issuecomment-335145885>,
or mute the thread

**Pull request merged**